src/representations/calculator_spherical_invariants.hh

```diff
@@ -401,18 +318,19 @@ namespace rascal {
 
     CalculatorSphericalExpansion rep_expansion;
 
-    internal::SphericalInvariantsType spherical_invariants_type{};
-    //! collection of precomputation for the different body order
-    std::array<std::shared_ptr<internal::SphericalInvariantsPrecomputationBase>,
-               internal::enumSize<internal::SphericalInvariantsType>()>
-        precompute_spherical_invariants{};
-    std::string spherical_invariants_type_str{};
+    internal::SphericalInvariantsType type{};
+
+    // precomputed l-factors the PowerSpectrum
+    Eigen::VectorXd l_factors{};
+
+    // precomputed wigner symbols for the BiSpectrum
+    Eigen::ArrayXd wigner_w3js{};
   };
 
   template <class StructureManager>
   void CalculatorSphericalInvariants::compute(StructureManager & managers) {
     using internal::SphericalInvariantsType;
-    switch (this->spherical_invariants_type) {
+    switch (this->type) {
```

Comment:
I prefer the verbose name.

Reply:
I don't see the point in repeating the local context: we are in the sperical_invariant class, and checking the value against `SphericalInvariantsType::XXX`, so I find this name clear enough.